src/pandoragroup/mmapper2group.cpp

```diff
@@ -564,7 +564,7 @@ void Mmapper2Group::slot_parseGmcpInput(const GmcpMessage &msg)
 
     if (msg.isCharVitals()) {
         // "Char.Vitals {\"hp\":100,\"maxhp\":100,\"mana\":100,\"maxmana\":100,\"mp\":139,\"maxmp\":139}"
-        QJsonDocument doc = QJsonDocument::fromJson(msg.getJson()->toQString().toUtf8());
+        const QJsonDocument &doc = msg.getDocument();
```

Comment:
Is there a reason we decided to pre-parse the JSON? The main reason I did it in a lazy fashion was because MMapper doesn't care about all GMCP messages and we could avoid some deserialization overhead.

Also, if MMapper only deserializes the message once then we aren't really gaining anything by doing it ahead of time. I assume this is the case here.

Reply:
The `GmcpMessage` class originally (i.e. before my changes) only contained unparsed JSON inside an UTF-8 string,
and parsed it every time it needed the JSON.

I first changed to it lazily parse the UTF-8 string to JSON on demand, but then I thought about multi-thread synchronization issues - i.e. thread safety - and finally I decided to just parse the UTF-8 string to JSON when GmcpMessage is created.

If you think parsing UTF-8 to JSON is a potentially expensive operation, and it's worth skipping it for messages MMapper is not interested into, I will put back the lazy parsing and its accompanying synchronization.

Reply:
[UPDATE] I modified `GmcpMessage` to lazily parse string to JSON